app/cdap/components/PipelineDiff/DiffList/index.tsx

```diff
@@ -0,0 +1,100 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import List from '@material-ui/core/List';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import Paper from '@material-ui/core/Paper';
+
+import styled from 'styled-components';
+
+import { ConnectionDiffListItem, PluginDiffListItem } from './DiffListItem';
+import { useAppSelector } from '../store/hooks';
+import {
+  getPluginNameFromStageDiffKey,
+  getStageDiffKeysFromConnectionDiffKey,
+  getStageNameFromStageDiffKey,
+} from '../util/diff';
+import { getAvailabePluginsMapKeyFromPlugin, getCustomIconSrc } from '../util/helpers';
+import { getPluginIcon } from 'services/helpers';
+
+const DiffListRoot = styled(Paper)`
+  &.MuiPaper-root {
+    height: 100%;
+    width: 400px;
+  }
+`;
+
+export const DiffList = () => {
+  const { diffMap, availablePluginsMap } = useAppSelector((state) => {
+    return {
+      diffMap: state.pipelineDiff.diffMap,
+      availablePluginsMap: state.pipelineDiff.availablePluginsMap,
+    };
+  });
+  return (
+    <DiffListRoot elevation={3}>
+      <List dense={true}>
+        {/* TODO: i18n */}
+        <ListSubheader>Plugins</ListSubheader>
+        {Object.keys(diffMap.stages).map((stageKey) => {
+          // The stage must exist in either the second pipeline(current version)
+          // or the first pipeline (older version).
+          const stage = diffMap.stages[stageKey].stage2 ?? diffMap.stages[stageKey].stage1;
+          return (
+            <PluginDiffListItem
+              nodeName={getStageNameFromStageDiffKey(stageKey)}
+              customIconSrc={getCustomIconSrc(
+                availablePluginsMap,
+                getAvailabePluginsMapKeyFromPlugin(stage.plugin)
+              )}
+              iconName={getPluginIcon(getPluginNameFromStageDiffKey(stageKey))}
+              diffKey={stageKey}
+              diffType={diffMap.stages[stageKey].diffIndicator}
+              key={stageKey}
+            />
+          );
+        })}
+        {/* TODO: i18n */}
+        <ListSubheader>Connections</ListSubheader>
+        {Object.keys(diffMap.connections).map((connectionKey) => {
+          const [fromStageKey, toStageKey] = getStageDiffKeysFromConnectionDiffKey(connectionKey);
+          const fromStage = diffMap.connections[connectionKey].from;
+          const toStage = diffMap.connections[connectionKey].to;
+          return (
+            <ConnectionDiffListItem
+              fromNodeName={getStageNameFromStageDiffKey(fromStageKey)}
```

Comment:
can you pull out some of these names into the consts above to make this a little cleaner.